### PR TITLE
Add continuous benchmark

### PR DESCRIPTION
**Pull request by pradithya**
This PR add continuous benchmark validation and report publishing using https://github.com/benchmark-action/github-action-benchmark.  
- Every pull request will be checked against performance regression and will fail if the performance degrade by more than 2x.
- Once a PR is merge to master, a benchmark report will be created and published in https://caraml-dev.github.io/universal-prediction-interface/dev/bench/

### CODE REVIEW

**Review by leonlnj — APPROVED** [review]
previously we were benchmarking all the various python version, now we are doing `3.7` only. Is this intended? If so what is the rationale for `3.7`? 

Left a question, other than that LGTM! Thanks!

File: .github/workflows/publish-benchmark.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: Publish Benchmark
+on:
+  push:
+    branches:
+      - master
```
Comment (tiopramayudi) [inline .github/workflows/publish-benchmark.yaml:5]: should it be `main`

File: .github/workflows/publish-benchmark.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: Publish Benchmark
```
Comment (tiopramayudi) [inline .github/workflows/publish-benchmark.yaml:1]: what is the difference between this and the `benchmark` in ci.yml?

**Review by tiopramayudi — COMMENTED** [review]
overall is LGTM

File: .github/workflows/publish-benchmark.yaml (hunk 1 above)
Comment (pradithya) [reply .github/workflows/publish-benchmark.yaml:5]: Good eye! Updated

File: .github/workflows/publish-benchmark.yaml (hunk 2 above)
Comment (pradithya) [reply .github/workflows/publish-benchmark.yaml:1]: The same benchmark, but the result is published to github pages https://caraml-dev.github.io/universal-prediction-interface/dev/bench/

**Comment by pradithya** [comment]
> previously we were benchmarking all the various python version, now we are doing 3.7 only. Is this intended? If so what is the rationale for 3.7?

Yeah, turns out it slow down the CI by too much. No particular reason of choosing 3.7, we can pick any version since it compare relative performance.

**Pull request merged**